### ISSUE TITLE
Some tests inside the test_astrometry_net_remote.py may need fixing. 

### ISSUE THREAD

**Issue by prajwel** [astropy/astroquery#2688]
I ran the function `test_solve_by_source_list()` from [test_astrometry_net_remote.py](https://github.com/astropy/astroquery/blob/main/astroquery/astrometry_net/tests/test_astrometry_net_remote.py). Some key values from astrometry.net provided WCS header do not match the `test-wcs-sol.fit` header key values.  This causes the test to fail. The key value differences and ratios are provided below. 

```
key, result[key], expected_result[key] - result[key], expected_result[key] / result[key]
SIMPLE, True, difference = 0, ratio = 1.0
BITPIX, 8, difference = 0, ratio = 1.0
NAXIS, 0, difference = 0, ratio = nan
EXTEND, True, difference = 0, ratio = 1.0
WCSAXES, 2, difference = 0, ratio = 1.0
EQUINOX, 2000.0, difference = 0.0, ratio = 1.0
LONPOLE, 180.0, difference = 0.0, ratio = 1.0
LATPOLE, 0.0, difference = 0.0, ratio = nan
CRVAL1, 87.4807133544, difference = 3.6489000052597476e-06, ratio = 1.0000000417109083
CRVAL2, 46.8397772451, difference = 8.982689999470495e-05, ratio = 1.000001917748232
CRPIX1, 2055, difference = 0, ratio = 1.0
CRPIX2, 2048.5, difference = 0.0, ratio = 1.0
CD1_1, 4.69538697321e-05, difference = -1.5093289000011625e-09, ratio = 0.9999678550690579
CD1_2, -0.000149104569833, difference = -2.235252200000204e-08, ratio = 1.000149911716489
CD2_1, 0.00014910882369, difference = -5.81169700000295e-09, ratio = 0.9999610237888263
CD2_2, 4.69652533611e-05, difference = 1.2625335700003056e-08, ratio = 1.000268822901964
IMAGEW, 4109, difference = 0, ratio = 1.0
IMAGEH, 4096, difference = 0, ratio = 1.0
A_ORDER, 2, difference = 0, ratio = 1.0
A_0_0, 0, difference = 0, ratio = nan
A_0_1, 0, difference = 0, ratio = nan
A_0_2, 2.46930846297e-08, difference = -1.400981001327e-07, ratio = -4.673576316350319
A_1_0, 0, difference = 0, ratio = nan
A_1_1, 2.89068901955e-08, difference = 1.0483714627550001e-07, ratio = 4.626718251824275
A_2_0, 6.61241319611e-09, difference = 2.013162670719e-08, ratio = 4.0445203755617065
B_ORDER, 2, difference = 0, ratio = 1.0
B_0_0, 0, difference = 0, ratio = nan
B_0_1, 0, difference = 0, ratio = nan
B_0_2, -1.2954576392e-07, difference = -1.54359570865e-07, ratio = 2.1915447189791832
B_1_0, 0, difference = 0, ratio = nan
B_1_1, 5.94272378319e-08, difference = 4.372024812441e-07, ratio = 8.356937613031944
B_2_0, -6.74838233729e-08, difference = -2.8200739146310005e-07, ratio = 5.178888767235259
AP_ORDER, 2, difference = 0, ratio = 1.0
AP_0_0, -1.53211029308e-08, difference = 6.033261998698e-07, ratio = -38.37877074482241
AP_0_1, -1.68343850658e-08, difference = 3.887530909198e-07, ratio = -22.092800206261987
AP_0_2, -2.46930743857e-08, difference = 1.400976681057e-07, ratio = -4.67356117417408
AP_1_0, -4.51042027647e-09, difference = -3.2852992334453e-07, ratio = 73.83798475685465
AP_1_1, -2.89068903344e-08, difference = -1.0483702191560001e-07, ratio = 4.626713932312568
AP_2_0, -6.61240994213e-09, difference = -2.013150895337e-08, ratio = 4.044504065772608
BP_ORDER, 2, difference = 0, ratio = 1.0
BP_0_0, 7.47161488383e-08, difference = 1.0932186277817e-06, ratio = 15.631624418271794
BP_0_1, 1.20431951512e-07, difference = 7.07423600012e-07, ratio = 6.874052451450239
BP_0_2, 1.29545708709e-07, difference = 1.5435870477800002e-07, ratio = 2.191538541232097
BP_1_0, -5.01839391266e-08, difference = -8.818790803123999e-07, ratio = 18.57293460140039
BP_1_1, -5.94272236725e-08, difference = -4.3720220968950006e-07, ratio = 8.356934796397292
BP_2_0, 6.74838234139e-08, difference = 2.8200776640609996e-07, ratio = 5.178894320738995
```

The same problem may also cause all other tests to fail, except `test_solve_image_upload_expected_failure`. 

We can try the following to mitigate the problem: 
* Update the test WCS files regularly.  
* Use `np.isclose` to check the difference.
* Other ideas?

**Comment by bsipocz** [comment]
Hmm, unfortunately, this is a side effect of not being able to run the tests from CI, or locally by the maintainers due to the requirement for an API key.
Ultimately it would be nice to be able to run tests without this limitation, even if it comes with some limited accessibility.

So for now, your suggestions make the most sense as workarounds. Please do use float comparison, numpy.testing contains a few useful functions for that in addition to `isclose`. And if necessary we could also update the test file.

**Comment by bsipocz** [comment]
(I labeled this as bug, as we should not literarily assert on float equality)

**Comment by prajwel** [comment]
Astrometry.net API is easy to generate; maybe one can be generated for running the tests. The API can be stored secretly for using it on the GitHub CI (https://docs.github.com/en/actions/security-guides/encrypted-secrets?tool=webui#using-encrypted-secrets-in-a-workflow). Perhaps it is best to find come up with a protocol for such situations as you have suggested in #2367.

**Comment by bsipocz** [comment]
That would be nice to be able to store the key on github. I gave a go for a keyring backend for CI, but didn't got too far with it as it was quite low priority to spend much time on: https://github.com/astropy/astroquery/pull/2396

**Comment by mwcraig** [comment]
This is closed by #2752
